docs/content/preview/manage/ysql-major-upgrade-local.md

```diff
@@ -0,0 +1,237 @@
+
+<ul class="nav nav-tabs-alt nav-tabs-yb">
+  <li>
+    <a href="../ysql-major-upgrade-yugabyted/" class="nav-link">
+      <img src="/icons/database.svg" alt="Server Icon">
+      Yugabyted
+    </a>
+  </li>
+
+  <li>
+    <a href="../ysql-major-upgrade-local/" class="nav-link active">
+      <i class="icon-shell"></i>
+      Local
+    </a>
+  </li>
+
+</ul>
+
+## Before you begin
+
+{{< warning >}}
+v2.25 is a preview release that is only meant for evaluation purposes and should not be used in production.
+{{< /warning >}}
+
+- All DDL statements, except ones related to Temporary table and Refresh Materialized View, are blocked for the duration of the upgrade. Consider executing all DDLs before the upgrade, and pause any jobs that might run DDLs.
+- Upgrade client drivers.
+
+    Upgrade all application client drivers to the new version. The client drivers are backwards compatible, and work with both the old and new versions of the database.
+- Your cluster must be running v2024.2.2.0 or later.
+
+    Deploy a new YugabyteDB cluster on version v2024.2.2.0 or later. If you have a pre-existing cluster, first upgrade it to the latest version in the v2024.2 series using the [upgrade instructions](../upgrade-deployment/).
+
+### Precheck
+
+New PostgreSQL major versions add many new features and performance improvements, but also remove some older unsupported features and data types. You can only upgrade after you remove all deprecated features and data types from your databases.
+
+Use the [pg_upgrade](https://www.postgresql.org/docs/15/pgupgrade.html) tool provided with v2.25.1.0 to make sure your cluster is compatible with the new version.
+
+Use the `--check` option as follows:
+
+```sh
+pg_upgrade --check -U <superuser_name> --old-host <any_yb_node_address>  --old-port <port> --old-datadir <yb_data_dir>
+```
+
+For example:
+
+```sh
+$ ./path_to_new_version/postgres/bin/pg_upgrade --check -U yugabyte --old-host 127.0.0.1 --old-port 5433 --old-datadir ~/yugabyte-data/node-1/disk-1/pg_data_11
+```
+
+```output
+Performing Consistency Checks on Old Live Server
+------------------------------------------------
+Checking cluster versions                                   ok
+Checking attributes of the 'yugabyte' user                  ok
+Checking for all 3 system databases                         ok
+Checking database connection settings                       ok
+Checking for system-defined composite types in user tables  ok
+Checking for reg* data types in user tables                 ok
+Checking for user-defined postfix operators                 ok
+Checking for incompatible polymorphic functions             ok
+Checking for invalid "sql_identifier" user columns          ok
+
+*Clusters are compatible*
+```
+
```

Comment:
The normal upgrade page has `1. Prepare the cluster for the upgrade` and `2. Download and install the new version`. Can we point to those sections from here?